examples/cube/main.rs

```diff
@@ -167,16 +188,80 @@ impl<R: gfx::Resources> gfx_app::Application<R> for App<R> {
         encoder.clear(&self.bundle.data.out_color, [0.1, 0.2, 0.3, 1.0]);
         encoder.clear_depth(&self.bundle.data.out_depth, 1.0);
         self.bundle.encode(encoder);
+
+        if self.screenshot == Screenshot::Ready {
+            use gfx::memory::Typed;
+
+            let (w, h, _, _) = self.bundle.data.out_color.get_dimensions();
+            encoder.copy_texture_to_buffer_raw(
+                self.bundle.data.out_color.raw().get_texture(),
+                None,
+                gfx::texture::RawImageInfo {
+                    xoffset: 0,
+                    yoffset: 0,
+                    zoffset: 0,
+                    width: w,
+                    height: h,
+                    depth: 0,
+                    format: ColorFormat::get_format(),
+                    mipmap: 0,
+                },
+                self.download.raw(),
+                0
+            ).unwrap();
+	          println!("screenshot taken");
+            self.screenshot = Screenshot::Some;
+        }
     }
 
-    fn on_resize(&mut self, window_targets: gfx_app::WindowTargets<R>) {
+    fn on_resize_ext<F: gfx::Factory<R>>(&mut self,
+                                         factory: &mut F,
+                                         window_targets: gfx_app::WindowTargets<R>) {
```

Comment:
That's hacky but it's currently hard to get a hold on the factory with `gfx_app`